diff --git a/ios/DevTools/ExpoRequestInterceptorProtocol.swift b/ios/DevTools/ExpoRequestInterceptorProtocol.swift
index 4e8682e..0fe3aa4 100644
--- a/ios/DevTools/ExpoRequestInterceptorProtocol.swift
+++ b/ios/DevTools/ExpoRequestInterceptorProtocol.swift
@@ -164,9 +164,6 @@ public final class ExpoRequestInterceptorProtocol: URLProtocol, URLSessionDataDe
     totalBytesExpectedToSend: Int64
   ) {
-    guard let task = self.task else {
-      return
-    }
     if #available(iOS 15.0, tvOS 15.0, macOS 12.0, *), let delegate = task.delegate {
       delegate.urlSession?(session, task: task, didSendBodyData: bytesSent, totalBytesSent: totalBytesSent, totalBytesExpectedToSend: totalBytesExpectedToSend)
       return
